### PR TITLE
Improve Flatpak build

### PR DESCRIPTION
**Pull request by jaje**
Tilix now can work from flatpak sandbox.

- Shell runs outside sandbox (in host environment)
- Uses PATH from host
- Sometime it crashes

### CODE REVIEW

File: experimental/flatpak/com.gexperts.Tilix.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                                "sed -i 's/\\/usr\\/bin\\/tilix/tilix/g' /app/share/dbus-1/services/com.gexperts.Tilix.service"
+                        ],
+                        "build-options": {
+                                "build-args": ["--share=network"]
+                        },
+                        "sources": [
+                                {
+                                        "type": "git",
+                                        "url": "../../.git",
+                                        "branch": "master"
```
Comment (jaje) [inline experimental/flatpak/com.gexperts.Tilix.json]: This presumes that this pull is merged into master, if you want to try it change branch and/or url.

**Comment by jaje** [comment]
Issue: #582

**Comment by gnunn1** [comment]
Not sure how can I test this, looks like flatpak requires xattr for the --user switch now, ugh. Not keen on compiling a kernel with it enabled nor running builds as root.

**Comment by jaje** [comment]
The --user switch has nothing to do with wherever it runs as root or not (it never does), it only switch form shared (/var/lib/flatpak) to user (~/.local/share/flatpak) repo.
Btw I thing xattr support depends on filesystem, not kernel.

**Comment by gnunn1** [comment]
Thanks, adding user_xattr to fstab does indeed work. Now when I run build.sh the install.sh script fails as it cannot find xdg-desktop-menu. Is there a dependency missing?

```
...
Running: ./install.sh /app
Your system is missing command xdg-desktop-menu, please install xdg-utils
Error: module tilix: Child process exited with code 1
```

**Comment by jaje** [comment]
No, this happens when you don't change branch as I said in comment. If you use the build script without this change it will build Tilix from master branch, not from this pull. 
There are possible 2 solutions: 
1) clone it locally and merge into master
2) change "branch" to "flatpak" (in experimental/flatpak/com.gexperts.Tilix.json)

**Comment by gnunn1** [comment]
Thanks it is working now, however I'm having some issues with it running without crashing. Notably bringing up preferences for example causes it to quit. Having said that, this is a huge improvement over what I have now so I'm to go ahead and merge it.

Thanks for making the effort on this, I really appreciate it as I was stuck on this for awhile.

**Pull request merged**